inject.js

```diff
@@ -316,34 +332,39 @@ module.exports = function inject (blobs, set, name, opts) {
         })
       }
 
-      if (!peerId && waiting[id]) queue(id, -1)
+      const peerId = findPeerWithBlob(id)
 
       if (peerId) return get(peerId, id)
+      if (waiting[id]) queue(id, -1)
     },
     push: function (id, cb) {
       id = toBlobId(id)
       // also store the id to push.
       if (!isBlobId(id)) { return cb(new Error('invalid hash:' + id)) }
-
-      push[id] = push[id] || {}
       queue(id, -1)
-      set.add(id, cb)
+      blobPush.add(id, cb) // leads to update of push
     },
     pushed: function () {
       return pushed.listen()
     },
     createWants: function () {
-      return createWantStream(this.id)
+      const source = pullDefer.source()
+      blobPush.onReady(() => {
+        source.resolve(createWantStream(this.id))
+      })
+      return source
```

Comment:
This is the other important change. Don't start the wants stream till we have all the wants + pushes loaded.

TODO - check this is actually needed ! I suspect with change further up the file (with blobPush.updates) we can actually remove this

Reply:
hmmm yes but causes more problems with that async-suite which would need fixing

Reply:
not enough time to want to do this